### DISCUSSION TITLE
tried to change to the vscode image, but fail

### DISCUSSION THREAD

**Discussion by yuanliuus** [eclipse-theia/theia-cloud#294] [Q&A]
Hello, 

I was trying to replace the application image with vscode image. Theia cloud looks to generate URLs with the format of https://hostname/instance_ID, however, vscode is trying to find the resource files in the root directory like http://hostname/resource_files. (the correct location should be https://hostname/instance_id/resource_files)  
I wonder if there is a way to generate the session URL using subdomains like https://instant_id.hostname/ ?

Thanks,

Yuan

**Comment by jfaltermeier** [comment]
Hi, via the `hosts.usePaths` helm value (https://github.com/eclipsesource/theia-cloud-helm/tree/theia-cloud-0.10.0/charts/theia.cloud) we offer these two options atm:

https://workspace.domain.io/instance-id/#/workspace/path
https://domain.io/workspace/instance-id/#/workspace/path

So no, 
https://instance-id.domain.io/#/workspace/path
is not supported out of the box at the moment and would require a custom operator. 

I think the main reason why we went for the other two options initially is that the third one might require a wildcard certificate.

**Reply by yuanliuus** [reply]
Thank you! I would like to customize the operator. I wonder if there is any documentation I can look into as a reference? The documentation on the website seems lack some details.

Thanks!

**Reply by jfaltermeier** [reply]
We have not yet released the Operator as a library, so you would have to directly change the code. 

The Ingress Rules are created here: 
https://github.com/eclipsesource/theia-cloud/blob/71218ad373608da5c1a2f7ceffd2e74023d721b8/java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/impl/LazySessionHandler.java#L471

Build a new operator image can be done as described here: https://github.com/eclipsesource/theia-cloud/blob/main/documentation/Building.md

I am working on https://github.com/eclipsesource/theia-cloud/issues/296 atm, which might be helpful
